content/blog/2020/07/2020-07-30-winsw-yaml-support-2.adoc

```diff
@@ -0,0 +1,143 @@
+
+==== Validate Configurations on Startup
+
+In the current implementation, configurations are validated on demand. 
+So there may be failures due to bad configurations at the runtime. 
+In this project, I will update WinSW to validate configurations at the startup. 
+It is not implemented yet and will be in phase 3.
+
+==== YAML Configuration support for Extensions
+
+At the moment there are 2 internal plugins in WinSW. RunAwayProcessKiller and SharedDirectoryMapper. 
+We allow users to provide configurations for those plugins in the same XML and YAML configuration file which is used to configure WinSW. 
+At the moment XML is implemented. 
+I have started working on YAML configuration support for extensions and it will be finished by the end of phase 3.
+
+==== Key updates in Phase 2
+
+* YAML Configuration structure
+
+** Environment variables 
+
+*** Now users can provide environment variables as a sequence of dictionaries that contains name and value for environment variables.
+
+** TimeStamp values
+
+*** Users can specify timestamp values in the same manner used in XML
+
+**** 10 ms, 5 sec, 3 min
+
+* YAML configuration document was published. 
+You can find the document https://github.com/winsw/winsw/blob/master/doc/YamlConfigFile.md[here].
+
+* Extend the WinSW to support both XML and YAML
+
+==== Sample YAML Configuration File
+
+```yaml
+id: jenkins
+name: Jenkins
+description: This service runs Jenkins automation server.
+env:
+    -
+        name: JENKINS_HOME
+        value: '%LocalAppData%\Jenkins.jenkins'
+    -
+        name: LM_LICENSE_FILE
+        value: host1;host2
+executable: java
+arguments: >-
+    -Xrs -Xmx256m -Dhudson.lifecycle=hudson.lifecycle.WindowsServiceLifecycle
+    -jar "E:\Winsw Test\yml6\jenkins.war" --httpPort=8081
+log:
+    mode: rotate
+onFailure:
+    -
+        action: restart
+        delay: 10 sec
+    -
+        action: reboot
+        delay: 1 hour
+```
+
+=== New CLI
+
+From the New, CLI users can use options. 
+Especially users can specify the configuration file from the Command line.
+So you don't need to locate WInSW executable and configuration file in the same folder.  
+There are more details on the new CLI in my previous blog post.
+
+==== Key updates in phase 2
+
+* Remove the /redirect command
+
+* *testwait* command was removed and add the *wait* option to the *test* command.
+
+* *stopwait* command was removed and add the *wait* option to the *stop* command.
+
+== Future Works
+
+* YAML schema validation
+
+** YAML Configuration file will be validated with the JSON schema file.
+
+* XML Schema validation
+
+** XML configuration file will be validated with the XSD file. 
+I have started working on this feature and you can find the implementation in this https://github.com/winsw/winsw/pull/460[Pull Request].
+
+* YAML Configuration validate on startup
+
+* YAML support for Extensions
+
+== How to contribute
```

Comment:
Add a section with a link to the video and slides?
https://youtu.be/oSBvJwSXOVQ?t=2736

Reply:
Would be great! I updated that and thank you so much for reminding that!